### PR TITLE
Add permissions checks to refund

### PR DESCRIPTION
**Pull request by cintamani**
From: https://eaflood.atlassian.net/browse/RUBY-811

This sorts out permissions checks on the refund functionality

### CODE REVIEW

File: spec/support/shared_examples/abilities/agency_with_refund_examples.rb [hunk 1]
```diff
@@ -4,4 +4,8 @@
   it "should be able to view revoked reasons" do
     should be_able_to(:view_revoked_reasons, WasteCarriersEngine::RenewingRegistration)
   end
+
+  it "should be able to refund a payment" do
```
Comment (irisfaraway) [inline spec/support/shared_examples/abilities/agency_with_refund_examples.rb]: Can we also test that the finance users and the "below agency_with_refund examples" cannot do this?

Comment (cintamani) [reply spec/support/shared_examples/abilities/agency_with_refund_examples.rb]: Oh yes! Good spot :)

File: spec/support/shared_examples/abilities/below_agency_with_refund_examples.rb [hunk 2]
```diff
@@ -4,4 +4,8 @@
   it "should not be able to view revoked reasons" do
     should_not be_able_to(:view_revoked_reasons, WasteCarriersEngine::RenewingRegistration)
   end
+
+  it "should not be able to refund a payment" do
```
Comment (irisfaraway) [inline spec/support/shared_examples/abilities/below_agency_with_refund_examples.rb]: This example doesn't cover the finance user types.

Comment (cintamani) [reply spec/support/shared_examples/abilities/below_agency_with_refund_examples.rb]: I am not sure I get the structure of those tests yet 😅 Added more

**Pull request merged**